lib/cocoapods/installer/analyzer.rb

```diff
@@ -527,8 +527,8 @@ def generate_pod_targets(resolver_specs_by_target)
             end
 
             pod_targets.each do |target|
-              dependencies = transitive_dependencies_for_specs(target.specs.reject(&:test_specification?), target.platform, specs.map(&:spec)).group_by(&:root)
-              test_dependencies = transitive_dependencies_for_specs(target.specs.select(&:test_specification?), target.platform, specs.map(&:spec)).group_by(&:root)
+              dependencies = transitive_dependencies_for_specs(target.specs.reject(&:test_specification?).to_set, target.platform, specs.map(&:spec).to_set).group_by(&:root)
```

Comment:
this might be faster by breaking out target.specs.to_set then select/rejecting on the one set twice and only doing specs.map(&:spec).to_set once

Reply:
👍  can use `#partition`

Reply:
addressed, hoisted the partitioning into the `PodTarget`